p2p/protocol/head/head_test.go

```diff
@@ -1,12 +1,11 @@
-package head_test
+package head
```

Comment:
It would be great to keep tests in a separate package where possible. It 
* helps maintain encapsulation, and
* showcases how to use the API in tests.

For a case where some internal API needs to be exposed for testing, like `deriveProtocolID`, there is already a nice pattern to expose just enough for testing purposes which I find to work really well. This is exposing APIs in a separate `*_test.go` file that is in the `head` package then using that exposed function in `head_test` package. [Example](https://github.com/filecoin-project/index-provider/blob/main/engine/chunker/cached_chunker_api_test.go). This way we achieve best of both words: maintained encapsulation between implementation and testing, and avoid exporting unwanted API just to be able to unit test.